### PR TITLE
Added ability to provide artifact ID instead of purl.

### PR DESCRIPTION
**Pull request by t-persson**
### Applicable Issues

fixes: eiffel-community/etos#73

### Description of the Change

Add the ability to start ETOS using artifact ID instead of PackageURL string for more explicit IUT selection.
Also ran black on all files.

### Alternate Designs

I opted to keep the packageurl imeplementation instead of removing as it would be a huge breaking change.

### Benefits

We can start tests on a more explicit IUT, instead of relying on the package URL being unique.

### Possible Drawbacks

None that I can think of.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobias.persson@axis.com>

### CODE REVIEW

File: src/etos_api/routers/etos/schemas.py [hunk 1]
```diff
@@ -36,9 +37,24 @@ class StartEtosRequest(BaseModel):
     iut_provider: Optional[str] = os.getenv("DEFAULT_IUT_PROVIDER", "default")
     log_area_provider: Optional[str] = os.getenv("DEFAULT_LOG_AREA_PROVIDER", "default")
 
+    @validator("artifact_id", always=True)
+    def validate_id_or_identity(cls, v, values):
```
Comment (fredjn) [inline src/etos_api/routers/etos/schemas.py:41]: Descriptive variable names FTW! V for Vendetta?

File: src/etos_api/routers/etos/utilities.py [hunk 2]
```diff
@@ -18,34 +18,49 @@
...
     """
     timeout = time.time() + timeout
     query_handler = GraphqlQueryHandler(etos_library)
+    if artifact_id is not None:
+        LOGGER.info("Verify that artifact ID %r exists.", artifact_id)
+        query = VERIFY_ARTIFACT_ID_EXISTS
+    else:
+        LOGGER.info("Getting artifact from packageURL %r", artifact_identity)
+        query = ARTIFACT_IDENTITY_QUERY
+    identity = artifact_identity or str(artifact_id)
```
Comment (fredjn) [inline src/etos_api/routers/etos/utilities.py:53]: Identity could easily be mixed up with artifact identity.
```suggestion
    artifact_identifier = artifact_identity or str(artifact_id)
```

File: src/etos_api/routers/etos/utilities.py [hunk 3]
```diff
@@ -18,34 +18,49 @@
...
+
     LOGGER.debug("Wait for artifact created event.")
     while time.time() < timeout:
         try:
-            artifact = await query_handler.execute(ARTIFACT_QUERY % artifact_identity)
+            artifact = await query_handler.execute(query % identity)
```
Comment (fredjn) [inline src/etos_api/routers/etos/utilities.py:58]:
```suggestion
            artifact_identifier = await query_handler.execute(query % identity)
```

File: src/etos_api/routers/etos/utilities.py [hunk 4]
```diff
@@ -18,34 +18,49 @@
...
             assert artifact is not None
             assert artifact["artifactCreated"]["edges"]
             return artifact["artifactCreated"]["edges"]
         except (AssertionError, KeyError):
             LOGGER.warning("Artifact created not ready yet")
         await asyncio.sleep(2)
-    LOGGER.error("Artifact %r not found.", artifact_identity)
+    LOGGER.error("Artifact %r not found.", identity)
```
Comment (fredjn) [inline src/etos_api/routers/etos/utilities.py:65]:
```suggestion
    LOGGER.error("Artifact %r not found.", artifact_identifier)
```

File: src/etos_api/routers/etos/utilities.py [hunk 5]
```diff
@@ -18,34 +18,49 @@
...
     :type timeout: int
     :return: ArtifactCreated edges from GraphQL.
     :rtype: list
     """
     timeout = time.time() + timeout
     query_handler = GraphqlQueryHandler(etos_library)
+    if artifact_id is not None:
+        LOGGER.info("Verify that artifact ID %r exists.", artifact_id)
+        query = VERIFY_ARTIFACT_ID_EXISTS
+    else:
```
Comment (fredjn) [inline src/etos_api/routers/etos/utilities.py:50]: I kind of dislike the implicitness of this. I'd rather see a elif with an explicit statment check for artifact_identity. This is not a review blocker though.

**Review by fredjn — CHANGES_REQUESTED** [review]

File: src/etos_api/routers/etos/utilities.py (hunk 3 above)
Comment (t-persson) [reply src/etos_api/routers/etos/utilities.py:58]: This is not an identifier. It is an artifact object. The identifier can be retrieved from the artifact though.

Comment (t-persson) [reply src/etos_api/routers/etos/utilities.py:58]: Correction: It's a response from GraphQL so it's edges and nodes etc. I.e. A list of artifacts.

**Review by fredjn — APPROVED** [review]

**Review by andjoe-axis — APPROVED** [review]

**Pull request merged**